lib/daru/index.rb

```diff
@@ -330,5 +341,11 @@ def values
     def inspect
       "Daru::MultiIndex:#{self.object_id} (levels: #{levels}\nlabels: #{labels})"
     end
+
+    def factor_out
+      # Remove levels not needed for display
+      useless = levels.index { |l| l.length > 1 }
+      drop_left_level useless
+    end
```

Comment:
> I don't quite understand how the correct number of levels are being dropped in spite of you nowhere specifying the number of levels the user has asked for.

`factor_out` method drops all levels from left until a level with size greater than 1 is encountered.
For example
```ruby
[17] pry(main)> d = Daru::MultiIndex.from_tuples([[:a, :one], [:a, :two]])
=> Daru::MultiIndex:20963520 (levels: [[:a], [:one, :two]]
labels: [[0, 0], [0, 1]])
```
Here `[:a]` level got dropped because it's size was 1. But there's a little problem with this. It's a little inconsistent. For example:
```ruby
[21] pry(main)> d = Daru::MultiIndex.from_tuples([[:a, :one], [:b, :two]])
=> Daru::MultiIndex:22630440 (levels: [[:a, :b], [:one, :two]]
labels: [[0, 1], [0, 1]])
[22] pry(main)> v = Daru::Vector.new([1,2], index: d)
=> 
#<Daru::Vector:22281380 @name = nil @size = 2 >
                  nil
[:a, :one]          1
[:b, :two]          2

[23] pry(main)> v[:a]
=> 1
```
Notice, it never showed the `[:one]`.

I checkout out the previous functionality and it too didn't showed `[:one]`.

If I want to correct this inconsistency I think I would need to pass arguments to the `factor_out` how much levels to remove. There's another way to do this by not involving passing arguments to the `factor_out` method and that is to save inside the Index object how much levels user wants to remove at the time of (indexes = @index[*indexes]) and use it to then display the result. Thoughts?